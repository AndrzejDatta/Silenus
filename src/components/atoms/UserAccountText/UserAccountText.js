import React from "react";
import PropTypes from "prop-types";
import { StyledUserAccountText } from "./UserAccountText.styles";
const UserAccountText = ({ text, isHeading, isUser, isLogout, isOption }) => {
  return (
    <StyledUserAccountText
      isHeading={isHeading}
      isOption={isOption}
      isLogout={isLogout}
      isUser={isUser}
    >
      {text}
    </StyledUserAccountText>
  );
};

UserAccountText.propTypes = {
  /**text in UserAccount */
  text: PropTypes.string.isRequired,
  /** bool which decides about font-weight */
  isHeading: PropTypes.bool.isRequired,
  /** bool which decides about color */
  isLogout: PropTypes.bool.isRequired,
};

export default UserAccountText;
