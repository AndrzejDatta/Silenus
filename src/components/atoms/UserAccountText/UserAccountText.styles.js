import styled from "styled-components";

export const StyledUserAccountText = styled.p`
  letter-spacing: 1px;
  line-height: 20px;
  font-size: ${({ theme }) => theme.font.size.big};
  color: ${(props) =>
    props.isHeadling || props.isOption
      ? ({ theme }) => theme.color.fontGreen
      : "black"};
  font-family: ${({ theme }) => theme.font.family.poppins};
  font-weight: ${(props) => (props.isHeadling ? "bold" : "normal")};
  padding: ${(props) =>
    props.isUser ? (props.isHeadling ? " 0" : "0px 8px") : "0 0 0 10px"}; ;
`;
