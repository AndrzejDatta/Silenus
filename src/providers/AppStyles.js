import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
