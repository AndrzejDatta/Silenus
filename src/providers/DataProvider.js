import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
