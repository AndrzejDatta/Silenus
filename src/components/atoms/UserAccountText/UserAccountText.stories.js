import React from "react";
import { storiesName } from "helpers/storiesName";
import UserAccountText from "components/atoms/UserAccountText/UserAccountText";

export default {
  title: `${storiesName.atoms}UserAccountText`,
  component: UserAccountText,
  argTypes: {
    isHeading: {
      options: [true, false],
      control: { type: "boolean" },
    },
    isUser: {
      options: [true, false],
      control: { type: "boolean" },
    },
  },
};

const Template = (args) => <UserAccountText {...args} />;

export const Default = Template.bind({});

Default.args = {
  text: "Text",
  isHeading: false,
  isLogout: false,
  color: "black",
};
