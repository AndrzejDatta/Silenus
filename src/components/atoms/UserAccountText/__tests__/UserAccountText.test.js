import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import { render, screen, rerender } from "@testing-library/react";
import UserAccountText from "../UserAccountText";

describe("Test UserAccountText component", () => {
  it("test default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <UserAccountText />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("font-weight", "normal");
    expect(tree).toHaveStyleRule("padding", "0 0 0 10px");
  });

  it("test UserAccountText as Headline + User", () => {
    render(
      <AppStyles>
        <UserAccountText isHeading={true} isUseInUserAccountPage={true} />
      </AppStyles>
    );
    expect(screen.getByText(/text/i)).toHaveStyleRule("padding", "0");
    expect(screen.getByText(/text/i)).toHaveStyleRule("font-weight", "bold");
  });
});
