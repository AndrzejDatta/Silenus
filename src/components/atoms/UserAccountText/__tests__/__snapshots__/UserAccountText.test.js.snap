// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserAccountText component test default style 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 20px;
  font-size: 16px;
  color: black;
  font-family: "Poppins",sans-serif;
  font-weight: normal;
  padding: 0 0 0 10px;
  width: 100%;
}

<p
  className="c0"
  color="black"
>
  text
</p>
`;
